### PR TITLE
Remove the Sort module.

### PR DESCRIPTION
**Pull request by whitequark**
See https://caml.inria.fr/mantis/view.php?id=7812.

It has been deprecated for 17 years (since 2001), and Sort.merge is documented to have undefined behavior when the lists being merged are not sorted in the first place.

### CODE REVIEW

**Review by xavierleroy — APPROVED** [review]
It's not just that the Sort module has been documented as deprecated for a long time; also, the Sort functions have triggered a deprecation warning since 4.02.  That's long enough to conclude that it's time to remove the Sort module.

**Comment by nojb** [comment]
There is one more use of `Sort.list` in https://github.com/ocaml/ocaml/blob/trunk/testsuite/tests/tool-lexyacc/output.ml#L47. Once removed, I think this can be merged.

**Comment by nojb** [comment]
The functional argument passed to `List.sort` in https://github.com/ocaml/ocaml/blob/trunk/testsuite/tests/tool-lexyacc/output.ml#L47 needs to be adjusted to return an `int` rather than a `bool`.

**Comment by whitequark** [comment]
@nojb Thanks! Fixed (in the right direction, I think)

File: manual/manual/cmds/comp.etex [hunk 1]
```diff
@@ -271,12 +271,12 @@ are two ways to fix this error:
 type (without type variables) to \var{name}. For instance, instead of
 writing
 \begin{verbatim}
-    let sort_int_list = Sort.list (<)
+    let sort_int_list = List.sort (<)
```
Comment (damiendoligez) [inline manual/manual/cmds/comp.etex]: This will give a type error. You should write: `List.sort Stdlib.compare` instead.

Same thing a few lines down.

**Review by damiendoligez — CHANGES_REQUESTED** [review]

File: manual/manual/cmds/comp.etex (hunk 1 above)
Comment (whitequark) [reply manual/manual/cmds/comp.etex]: Fixed.

**Comment by alainfrisch** [comment]
@damiendoligez Since your comment has been addressed, can you accept the PR?

**Comment by damiendoligez** [comment]
I'm running an OPAM check on this PR (mostly out of curiosity). Let's wait a few more hours for the results.

**Review by damiendoligez — APPROVED** [review]

**Comment by damiendoligez** [comment]
FTR: this PR doesn't break any of the packages that currently compile with trunk.

**Pull request merged**